src/KubernetesClient/KubernetesClientConfiguration.ConfigFile.cs

```diff
@@ -557,19 +558,32 @@ public static ExecCredentialResponse ExecuteExternalCommand(ExternalExecution co
                 throw new KubeConfigException($"external exec failed due to: {ex.Message}");
             }
 
-            var stdout = process.StandardOutput.ReadToEnd();
-            var stderr = process.StandardError.ReadToEnd();
-            if (string.IsNullOrWhiteSpace(stderr) == false)
+
+            var stdOutTask = process.StandardOutput.ReadToEndAsync(); // Assumes process exits
+
+            var buffer = new char[4096];
+            var stdErrorTask = process.StandardError.ReadAsync(buffer, 0, buffer.Length); // Assumes process will continue
+
+            var result = Task.WaitAny(new Task[] { stdErrorTask, stdOutTask }, TimeSpan.FromSeconds(10));
+
+            if (result == -1)
             {
-                throw new KubeConfigException($"external exec failed due to: {stderr}");
+                throw new KubeConfigException("external exec failed due to timeout");
             }
+            else if (result == 0)
+            {
+                var stderrBuilder = new StringBuilder();
+                stderrBuilder.Append(buffer, 0, stdErrorTask.Result);
 
-            // Wait for a maximum of 5 seconds, if a response takes longer probably something went wrong...
-            process.WaitForExit(5);
+                if (stderrBuilder.Length > 0)
```

Comment:
here my suggestion is to ignore stderr
throw only when cannot deserialize the stdout

the exec may write some debug log to stderr

Reply:
I think logging of non errors to stderr is a bad practice.

My testing of kubelogin shows the following workflow, and I modeled the current logic to follow.

![kubelogin drawio](https://user-images.githubusercontent.com/9521987/234185959-5c6e0978-6ff6-41f6-8399-a7917b51b3fd.png)


Also looking at the google auth plugin appears to be similar, errors are only printed on error conditions.
https://github.com/kubernetes/cloud-provider-gcp/blob/master/cmd/gke-gcloud-auth-plugin/cred.go#L145

Reply:
just double checked client-go

https://github.com/kubernetes/client-go/blob/6c514ab3ffc5f500fd373f09956780c0e3bbe028/plugin/pkg/client/auth/exec/exec.go#L450

the impl is to ignore and redirect stderr to os.stderr and deserialize the stdout directly

Reply:
I'm not sure how to ignore the stderr as we need its data, and also block the call and wait for stdout. The current model works well with my testing.
See recording [Video.webm](https://user-images.githubusercontent.com/9521987/234685591-bc3a70de-0099-4355-bb9c-547c022ca3bb.webm)

Any suggestions?

Reply:
ignore here means do not `if (stderrBuilder.Length > 0)`
just leave it there, only throw it if  cannot deserialize stdout

BTW this kubeui is so cool